src/org/usfirst/frc/team2706/robot/commands/teleop/GearHandler.java

```diff
@@ -0,0 +1,34 @@
+package org.usfirst.frc.team2706.robot.commands.teleop;
+
+import org.usfirst.frc.team2706.robot.Robot;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.command.Command;
+
+public class GearHandler extends Command {
+
+    private DoubleSolenoid solenoid = new DoubleSolenoid(4,5);
+    private Joystick stick ;
+    private boolean closed;
+    
+    public GearHandler(boolean closed) {
+        stick = Robot.oi.getDriverJoystick();
+        this.closed = closed;
+    }
+    
+    protected void execute(){
+        if (stick.getRawButton(2)){
```

Comment:
Our new code model requests that any joystick calls be put in OI, so you should change the command to be controlStick.whenPressed(new GearHandler(closed)); in OI and get rid of this line